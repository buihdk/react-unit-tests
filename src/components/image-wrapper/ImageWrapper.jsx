import React from 'react';
import PropTypes from 'prop-types';
import I18n from 'i18n-js';

const styles = {};

const ImageWrapper = props => {
  if (props.src) {
    return (
      <div className={styles.imageWrapper}>
        <img className={styles.image} src={props.src} alt="imageAd" />
      </div>
    );
  }
  return (
    <div className={styles.noImageWrapper}>
      <p>{I18n.t('common.noImage')}</p>
    </div>
  );
};

ImageWrapper.propTypes = {
  src: PropTypes.string,
};

ImageWrapper.defaultProps = {
  src: '',
};

export default ImageWrapper;
