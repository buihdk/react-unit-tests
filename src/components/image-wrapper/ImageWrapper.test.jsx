import React from 'react';
import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ImageWrapper from './ImageWrapper';

describe('Test Image Wrapper', () => {
  const props = {
    src: 'ABCDXYZ',
  };

  test('renders without crashing', () => {
    const wrapper = shallow(<ImageWrapper />, true);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('render with props', () => {
    const wrapper = mount(<ImageWrapper {...props} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
