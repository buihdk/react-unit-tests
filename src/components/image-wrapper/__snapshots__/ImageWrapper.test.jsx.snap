// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Image Wrapper render with props 1`] = `
<ImageWrapper
  src="ABCDXYZ"
>
  <div>
    <img
      alt="imageAd"
      src="ABCDXYZ"
    />
  </div>
</ImageWrapper>
`;

exports[`Test Image Wrapper renders without crashing 1`] = `
<div>
  <p>
    No image
  </p>
</div>
`;
